src/redisearch_rs/inverted_index_bencher/src/bencher.rs

```diff
@@ -96,51 +45,314 @@ impl NumericBencher {
     }
 
     pub fn encoding(&self, c: &mut Criterion) {
-        for (group, values) in &self.encoding_test_values {
-            let group = format!("Encode - {}", group);
+        for BenchGroup { group, inputs } in &self.test_values {
+            let group = format!("Encode | {}", group);
             let mut group = self.benchmark_group(c, &group);
-            numeric_c_encode(&mut group, values);
+
+            for input in inputs {
+                numeric_c_encode(&mut group, input);
+                numeric_rust_encode(&mut group, input);
+            }
+
             group.finish();
         }
     }
 
     pub fn decoding(&self, c: &mut Criterion) {
-        for (group, values) in &self.decoding_test_values {
-            let group = format!("Decode - {}", group);
+        for BenchGroup { group, inputs } in &self.test_values {
+            let group = format!("Decode | {}", group);
             let mut group = self.benchmark_group(c, &group);
-            numeric_c_decode(&mut group, values);
+
+            for input in inputs {
+                numeric_c_decode(&mut group, input);
+                numeric_rust_decode(&mut group, input);
+            }
+
             group.finish();
         }
     }
 }
 
-fn numeric_c_encode<M: Measurement>(group: &mut BenchmarkGroup<'_, M>, values: &[f64]) {
-    group.bench_function("C", |b| {
-        b.iter_batched(
-            || TestBuffer::with_capacity(64),
-            |mut buffer| {
-                for &value in values {
-                    let mut record = inverted_index::RSIndexResult::numeric(value);
-                    let grew_size = encode_numeric(&mut buffer, &mut record, 684);
+#[derive(Clone)]
+struct BenchEncodingInputs<'a> {
+    group: &'a str,
+    values: Vec<f64>,
+    value_size_fn: fn(f64) -> usize,
+}
+
+struct BenchGroup {
+    group: String,
+    inputs: Vec<BenchInput>,
+}
+
+struct BenchInput {
+    /// Stores the value, delta and the encoded buffer for this benchmark run
+    values: Vec<(f64, usize, Vec<u8>)>,
+    delta_size: usize,
+    value_size: usize,
+}
+
+fn generate_test_values() -> Vec<BenchGroup> {
```

Comment:
```suggestion
fn numeric_generate_test_values() -> Vec<BenchGroup> {
```

Best to namespace it as each encoder will require different test values.

Reply:
In general I'm pondering how much of this will be shared with other benchers.
Let me finish my implementation and we can then consider how they should all fit together.

Reply:
Yeah, it might be better to rename this file to `numeric_bencher` or `benchers/numeric`